### PR TITLE
BREAKING(unstable): remove Deno.SymlinkOptions

### LINKED ISSUE #7234: Remove Deno.SymlinkOptions
Currently we expose Deno.SymlinkOptions which just a leak of an internal implementation for Windows.
I propose we remove it while the ops using it are still unstable.

Also, seems we have zero coverage for it so who knows if it works.

### PR DESCRIPTION
**Pull request by caspervonb**
This removes Deno.SymlinkOptions which is leaking internal implementation details on Windows which encourages checking and special casing when running on Windows.

Closes #7234

### CODE REVIEW

**Comment by caspervonb** [comment]
Defaulting to file for non-existing paths aligns with Go's implementation https://golang.org/src/os/file_windows.go#347

File: cli/ops/fs.rs [hunk 1]
```diff
@@ -1280,30 +1271,16 @@ fn op_symlink_sync(
...
-            } else if metadata.is_dir() {
-              symlink_dir(&oldpath, &newpath)?
-            }
-          }
-          Err(_) => return Err(ErrBox::type_error(
-            "you must pass a `options` argument for non-existent target path in windows"
-              .to_string(),
-          )),
-        }
+    if let Ok(metadata) = std::fs::metadata(&oldpath) {
```
Comment (ry) [inline cli/ops/fs.rs:1274]: This introduces an extra syscall...

Comment (ry) [reply cli/ops/fs.rs:1274]: @piscisaureus are you ok with this change?

Comment (piscisaureus) [reply cli/ops/fs.rs:1274]: Not ok.
Auto detection requires the symlink target to exist which is not guaranteed.
Symlink types on windows are not an implementation detail.

Comment (caspervonb) [reply cli/ops/fs.rs:1274]: > Auto detection requires the symlink target to exist which is not guaranteed.

Hence doing the same as Go's implementation of assuming file when no target exists.

Comment (piscisaureus) [reply cli/ops/fs.rs:1274]: Well I know of these examples:
- node.js
- git (I had to add that myself because auto detection didn't work)
- rust

Comment (piscisaureus) [reply cli/ops/fs.rs:1274]: https://nodejs.org/api/fs.html#fs_fs_symlink_target_path_type_callback

Comment (caspervonb) [reply cli/ops/fs.rs:1274]: > https://nodejs.org/api/fs.html#fs_fs_symlink_target_path_type_callback

Fair enough, I'll consider Node a tie breaker.

**Pull request closed without merging**